redisinsight/api/src/modules/browser/services/string-business/string-business.service.ts

```diff
@@ -89,20 +94,51 @@ export class StringBusinessService {
       }
       catchAclError(error);
     }
+
     if (result.value === null) {
       this.logger.error(
         `Failed to get string value. Not Found key: ${keyName}.`,
       );
       throw new NotFoundException();
-    } else {
-      this.recommendationService.check(
-        clientMetadata,
-        RECOMMENDATION_NAMES.STRING_TO_JSON,
-        { value: result.value, keyName: result.keyName },
-      );
-      this.logger.log('Succeed to get string value.');
-      return plainToClass(GetStringValueResponse, result);
     }
+
+    this.recommendationService.check(
+      clientMetadata,
+      RECOMMENDATION_NAMES.STRING_TO_JSON,
+      { value: result.value, keyName: result.keyName },
+    );
+    this.logger.log('Succeed to get string value.');
+
+    return result;
+  }
+
+  public async getStringValue(
+    clientMetadata: ClientMetadata,
+    dto: GetStringInfoDto,
+  ): Promise<GetStringValueResponse> {
+    const { keyName, stringMaxLen } = dto;
+    const result: GetStringValueResponse = await this.getStringValueFromClient(
```

Comment:
I would recommend to have 2 separate methods like (getValue and getPartialValue, something like this) OR handle everything in a single function
But this is optional